SDWebImage/Core/SDAnimatedImagePlayer.m

```diff
@@ -297,7 +297,10 @@ - (void)displayDidRefresh:(SDDisplayLink *)displayLink {
         NSTimeInterval currentDuration = [self.animatedProvider animatedImageDurationAtIndex:currentFrameIndex];
         currentDuration = currentDuration / playbackRate;
         if (self.currentTime < currentDuration) {
-            // Current frame timestamp not reached, return
```

Comment:
Does this means, we always fetch the **next frame index** even the current time duration does not hit ?

This seems sounds reasonable. And this logic get effected by that `needsDisplayWhenImageBecomesAvailable`

Reply:
A question: what about the `first frame` ?

The first frame (poster image) has a specify logic that always set `needsDisplayWhenImageBecomesAvailable=YES`, so it render immediatelly without the duration.

After this commit, it will try to prefetch the `second frame` here, even if the first frame is available in cache. Sounds use the RAM to exchange the performance.

Reply:
![image](https://user-images.githubusercontent.com/6919743/209632525-c7836f78-f40a-4c74-9d57-6d71614f8710.png)

The `animatedImageDurationAtIndex:0` for poster image does not return 0, which will enter this logic :)

Reply:
Anway, I think it's OK for this. The prefetch seems safe and only has some RAM increment. Let me merge just now

Reply:
Thanks.